### ISSUE TITLE
Release Android app crashes on start. Says it can not find catalyst instance

### ISSUE THREAD

**Issue by TheRogue76** [software-mansion/react-native-reanimated#1969]
## Description

Android app crashes on first launch while in production mode but works fine afterwards. The LogCat shows that it was looking for a catalyst instance before crash

### Expected behavior

For the app to open correctly on first Launch

### Actual behavior & steps to reproduce

It crashes just once after installation and only on android.
Steps: 

1) install reanimated as per the documentation. (adding with yarn, add babel plugin, enable hermes, Add in MainApplication.Java and add proguard)
2) create a small collapsible Header with animation using ScrollView
3) cd android && ./gradlew assembleRelease
4) install on simulator or physical device
5) crash on first launch and get the following Errors:
```
Unhandled SoftException
java.lang.RuntimeException: Catalyst Instance has already disappeared: requested by Timing
	at com.facebook.react.bridge.ReactContextBaseJavaModule.getReactApplicationContextIfActiveOrWarn(Unknown Source:40)
	at com.facebook.react.modules.core.TimingModule.access$000(Unknown Source:0)
	at com.facebook.react.modules.core.TimingModule$a.callTimers(Unknown Source:2)
	at com.facebook.react.modules.core.d.q(Unknown Source:55)
	at com.facebook.react.modules.core.TimingModule.createTimer(Unknown Source:6)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.facebook.react.bridge.JavaMethodWrapper.invoke(Unknown Source:147)
	at com.facebook.react.bridge.JavaModuleWrapper.invoke(Unknown Source:21)
	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:0)
	at android.os.Looper.loop(Looper.java:214)
	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(Unknown Source:37)
	at java.lang.Thread.run(Thread.java:919)

JNI ERROR (app bug): accessed stale Global 0x65636166  (index 106313238 in a table of size 771)

java_vm_ext.cc:570] JNI DETECTED ERROR IN APPLICATION: use of deleted global reference 0x65636166

java_vm_ext.cc:570]     from void com.swmansion.reanimated.NativeProxy$EventHandler.receiveEvent(java.lang.String, com.facebook.react.bridge.WritableMap)
```

## Snack or minimal code example

```

const scrollY = useSharedValue<number>(0)

const headerHeight = useDerivedValue<number>(() => {
    return value based on header
  });

const heightStyle = useAnimatedStyle(() => {
    return {height: headerHeight.value};
});


const scrollHandler = useAnimatedScrollHandler(event => {
    gets scrollview position
});
.
.
.

<Animated.View
  style={[...,heightStyle]}>
  <Text>{title}</Text>
</Animated.View>
 <Animated.ScrollView onScroll={scrollHandler}>
{sth}
</Animated.ScrollView>
```
## Package versions

- React Native: 0.64.0
- React Native Reanimated: 2.1.0
- NodeJS: 12.18.2
- Xcode: 12.4/12D4e 
- Java & Gradle: Java 1.8.0_242 and Gradle 7

## Affected platforms

- [x] Android
- [] iOS
- [] Web

**Comment by TheRogue76** [comment]
Is there anyone tracking this issue? i have tried disabling proguard as well but the issue still persists in release builds.

**Comment by matinzd** [comment]
I have the same issue with iOS too but it stops working at first launch. 
Android behaves same.

**Comment by andrisundae** [comment]
There is no found solution yet .?

**Comment by TheRogue76** [comment]
> There is no found solution yet .?

No. I switched out to React Native's core animations in the meantime. Unless you are using it in a personal or relatively unimportant project i suggest you switch out for now as well. This issue has been open for a month.

**Comment by kacperkapusciak** [comment]
Hi @TheRogue76 :wave:

Could you confirm the crash still occurs on the newest version of `react-native-reanimated`?

If so, could you provide a higher-quality reproduction of the issue? Provided snippet references non-existent variables and values thus it isn't very helpful.

Cheers

**Comment by TheRogue76** [comment]
Hi @kacperkapusciak. I did eventually resolve the issue and as such, I will close both this and the other mentioned instance. In my case, the problem was one of my third-party modules (react-native-splash-screen) called a native method for a layer that did not exist and it crashed on its first launch. The application's final message before crashing was its catalyst instance disappearing which I have since learned is related to something else. I have worked with reanimated on other projects since then and can confirm that it works perfectly fine.
